### PR TITLE
feat(Add Redux): added redux

### PR DESCRIPTION
**Pull request by EmilyLip**
Added redux ans user model.
On Identification we will have to setUser.

### CODE REVIEW

File: client/src/redux/User/userActionCreatores.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import * as actionTypes from './userActionTypes';
+import {store} from '../store';
+import User from 'models/User';
```
Comment (MayaSasson) [inline client/src/redux/User/userActionCreatores.ts:3]: imports

File: client/src/redux/User/userReducer.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import User from 'models/User';
+import * as Actions from './userActionTypes';
+
+
+const initialState: User = {
+    id: 'XXXX',
+    name: 'XXXXXX'
```
Comment (MayaSasson) [inline client/src/redux/User/userReducer.ts:7]: Maybe you should use undefined or ''?

**Review by yaniryos — APPROVED** [review]

**Review by rockenmush — APPROVED** [review]

File: client/src/components/App/AppToolbar/TopToolbar/TopToolbar.tsx [hunk 3]
```diff
@@ -13,9 +15,17 @@ const TopToolbar: React.FC = (): JSX.Element => {
...
-export default TopToolbar;
+const mapStateToProps = (state: StoreStateType) => ({
+    userName: state.user.name
+});
+
+interface Props {
+    userName: string
+}
+
+export default connect(mapStateToProps)(TopToolbar);
```
Comment (RoeiRom) [inline client/src/components/App/AppToolbar/TopToolbar/TopToolbar.tsx:31]: You can use the useSelector hook from redux and remove the connect and the props mapping

File: client/src/components/App/Content/LandingPage/LandingPage.tsx [hunk 4]
```diff
@@ -1,11 +1,27 @@
...
 
-export default LandingPage;
+const mapStateToProps = (state: StoreStateType) => ({
+    user: state.user
+});
+
+interface Props {
+    user: User
+}
+
```
Comment (RoeiRom) [inline client/src/components/App/Content/LandingPage/LandingPage.tsx:26]: Same here

File: client/src/redux/User/userReducer.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+}
+
+const userReducer = (state = initialState, action: Actions.UserAction) : User => {
+    switch (action.type) {
```
Comment (RoeiRom) [inline client/src/redux/User/userReducer.ts:11]: Why spread the object

File: package.json [hunk 6]
```diff
@@ -21,5 +21,12 @@
   "homepage": "https://github.com/CoronaTeamMOD/CoronaInvestigation#readme",
   "devDependencies": {
     "concurrently": "^5.3.0"
+  },
+  "dependencies": {
+    "@reduxjs/toolkit": "^1.4.0",
+    "@types/react-redux": "^7.1.9",
```
Comment (RoeiRom) [inline package.json:27]: Move types to Dev dependancy

File: client/src/index.tsx [hunk 7]
```diff
@@ -2,17 +2,19 @@ import React from 'react';
 import ReactDOM from 'react-dom';
```
Comment (RoeiRom) [inline client/src/index.tsx:2]: Move down

File: client/src/redux/User/userReducer.ts (hunk 5 above)
Comment (RoeiRom) [reply client/src/redux/User/userReducer.ts:11]: You can only return the user

**Pull request merged**